### PR TITLE
IB 40 integrate backend with frontend

### PR DESCRIPTION
**Pull request by garcialucasm**
There are some changes I made to fix some bugs or remove code that became unnecessary. The main point to review is the folder /services/.

@felipedreis if you have any suggestions, please let me know.

### CODE REVIEW

File: frontend/src/components/atoms/StatusIndicator.tsx [hunk 1]
```diff
@@ -20,36 +20,34 @@ enum AnimationStatusClass {
...
   ) {
     statusColorClass = StatusColorClass.YELLOW;
     animationStatusClass = AnimationStatusClass.BOOKED;
   } else if (
-    status === BookingStatus.HANDEDOVER ||
-    status === BikeStatus.INUSE ||
-    status === UserStatus.INUSE
+    bookingStatus === BookingStatus.DELIVERED ||
+    bookingStatus === BikeStatus.INUSE ||
+    bookingStatus === UserStatus.INUSE
```
Comment (felipedreis) [inline frontend/src/components/atoms/StatusIndicator.tsx:38]: is this booking status comparision with the user status correct?

File: frontend/src/components/atoms/StatusIndicator.tsx [hunk 2]
```diff
@@ -20,36 +20,34 @@ enum AnimationStatusClass {
...
   let statusColorClass: string;
   let animationStatusClass: string;
+  let bookingStatus = status.toUpperCase();
 
-  if (
-    status === BookingStatus.RETURNED ||
-    status === BikeStatus.FREE ||
-    status === UserStatus.FREE
-  ) {
+  if (bookingStatus === BikeStatus.FREE || bookingStatus === UserStatus.FREE) {
```
Comment (felipedreis) [inline frontend/src/components/atoms/StatusIndicator.tsx:25]: Same here as in the comment bellow. You're comparing the `UserStatus` and `BikeStatus`enum with `bookingStatus` variable

File: frontend/src/services/bookingApi.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+//Url to Cancel a booking
+const cancelBookingUrl = "/booking/all";
+//Url to Approve a booking
+const approveBookingUrl = "/booking/approve/";
+//Url to Return a booking
+const returnBookingUrl = "/booking/return/";
+
+
+// Show all active bookings
+export const bookingFetchApi = async () => {
```
Comment (felipedreis) [inline frontend/src/services/bookingApi.ts:22]: any particular reason to not use just `export async function bookingFetchApi()`?

File: frontend/src/services/bookingApi.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        'Content-Type': 'application/json',
+      },
+    });
+
+    if (!response.ok) {
+      if (response.status === 404) {
+        throw new Error('Data not found');
+      } else if (response.status === 500) {
+        throw new Error('Server error');
+      } else {
```
Comment (felipedreis) [inline frontend/src/services/bookingApi.ts:143]: besides 404, 500, the API can return a 401 as described in IB-48. Could you handle this code as well?

Comment (felipedreis) [reply frontend/src/services/bookingApi.ts:143]: This code is valid for all routes, and for `/booking/all` there is also code 204 for empty responses.

File: frontend/src/types/BikeType.tsx [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
-  NONE = "none",
-  CLASSIC = "classic",
-  SMALL = "small",
-  STANDARD = "standard",
-  FREE = "free"
+  NONE = "NONE",
+  CLASSIC = "CLASSIC",
+  SMALL = "SMALL",
+  STANDARD = "STANDARD",
+  FREE = "FREE"
```
Comment (felipedreis) [inline frontend/src/types/BikeType.tsx:6]: Why do we have a `FREE` bike size?

**Review by felipedreis — APPROVED** [review]

File: frontend/src/components/atoms/StatusIndicator.tsx (hunk 1 above)
Comment (garcialucasm) [reply frontend/src/components/atoms/StatusIndicator.tsx:38]: It was a wrong nomenclature. I've already made a commit with the correction. Thanks.

File: frontend/src/components/atoms/StatusIndicator.tsx (hunk 2 above)
Comment (garcialucasm) [reply frontend/src/components/atoms/StatusIndicator.tsx:25]: It was a wrong nomenclature. I've already made a commit with the correction. Thanks.

File: frontend/src/services/bookingApi.ts (hunk 3 above)
Comment (garcialucasm) [reply frontend/src/services/bookingApi.ts:22]: I didn't notice this unnecessary const. Thanks for the advice. It's also in the same commit mentioned before.

File: frontend/src/services/bookingApi.ts (hunk 4 above)
Comment (garcialucasm) [reply frontend/src/services/bookingApi.ts:143]: Thank you for the advice. In IB-50 I changed the code to handle all returns at the same time. It's already in the main branch.

File: frontend/src/types/BikeType.tsx (hunk 5 above)
Comment (garcialucasm) [reply frontend/src/types/BikeType.tsx:6]: Great question. I've already also commit the correction. Thanks again.

**Pull request merged**